test_libs/pyspec/eth2spec/test/phase_0/epoch_processing/test_process_justification_and_finalization.py

```diff
@@ -147,16 +149,19 @@ def finalize_on_123(spec, state, epoch, support):
     # 3210x -- justification bitfield indices
     # 011*. -- justification bitfield contents, . = this epoch, * is being justified now
     # checkpoints for the epochs ago:
-    # c4 = spec.Checkpoint(epoch=epoch - 4, root=b'\xaa' * 32)
+    c4 = spec.Checkpoint(epoch=epoch - 4, root=b'\xaa' * 32)
     c3 = spec.Checkpoint(epoch=epoch - 3, root=b'\xaa' * 32)
     c2 = spec.Checkpoint(epoch=epoch - 2, root=b'\xbb' * 32)
     c1 = spec.Checkpoint(epoch=epoch - 1, root=b'\xcc' * 32)
+    state.block_roots[spec.get_epoch_start_slot(c3.epoch) % spec.SLOTS_PER_HISTORICAL_ROOT] = c3.root
+    state.block_roots[spec.get_epoch_start_slot(c2.epoch) % spec.SLOTS_PER_HISTORICAL_ROOT] = c2.root
+    state.block_roots[spec.get_epoch_start_slot(c1.epoch) % spec.SLOTS_PER_HISTORICAL_ROOT] = c1.root
 
     old_finalized = state.finalized_checkpoint
-    state.previous_justified_checkpoint = c3
+    state.previous_justified_checkpoint = c4
```

Comment:
It feels strange to do this as it is not the behaviour one would expect from this test. The problem, however, is that otherwise, a new checkpoint is finalised as the condition of "2nd/3rd most recent epochs are justified, the 2nd using the 3rd as source" is met. This means that in the case where there is insufficient support, `assert state.finalized_checkpoint == old_finalized  # no new finalized` fails. @djrtwo, please can you check me here (and propose an alternative solution if you have one.)

TLDR: this is required because of an artifact of how the tests are constructed by manipulating what is and isn't justified in the state.